winterfell/webapp/src/js/claimConfirmCtrl.js

```diff
@@ -1,87 +1,87 @@
       }
     ];
 
-    // list of forms user has filled out
-    // TODO: placeholder forms for now
-    $scope.formsList = [
-      {
-        id: 'va_2723',
-        title: 'VA-27343 Intent to File',
-        summary: 'Something that blah blah holds your place in line when filing.'
-      }
-    ];
+    $scope.formsList = forms;
+    $scope.formsInfo = formTemplateService;
 
-    $scope.onClickAddAddress = function() {
-      return openModifyContactModal('address', $scope.mailingList);
-    };
+    console.log($scope);
 
-    $scope.onClickEditAddress = function(index) {
-      return openModifyContactModal('address', $scope.mailingList, index);
+    $scope.onClickEditAddress = function (index) {
+      openModifyContactModal();
     };
 
-    $scope.onClickAddEmail = function() {
-      return openModifyContactModal('email', $scope.emailList);
+    $scope.onClickEditEmail = function (index) {
+      openModifyEmailModal();
     };
 
-    $scope.onClickEditEmail = function(index) {
-      return openModifyContactModal('email', $scope.emailList, index);
-    };
+    function openModifyContactModal() {
+      var modalInstance = $uibModal.open({
+        scope: $scope,
+        controller: 'claimConfirmModifyAddressCtrl',
+        templateUrl: 'templates/modals/confirmClaimModifyAddress.html',
+        resolve: {
+          address: function () {
+            return _.cloneDeep($scope.userAddress);
+          }
+        }
+      });
+
+      modalInstance.result.then(function (address) {
+        console.log("new address", address);
+        $scope.userAddress = address;
+      }, function () {
+        console.log('modal-component dismissed at: ' + new Date());
+      });
+    }
 
-    function openModifyContactModal(forType, list, index) {
-      $scope.modalData = {
-        forType: forType,
-        list: list,
-        targetIndex: index >= 0 ? index : -1
-      };
-      return $uibModal.open({
+    function openModifyEmailModal() {
+      var modalInstance = $uibModal.open({
         scope: $scope,
-        controller: 'claimConfirmModifyContactCtrl',
-        templateUrl: 'templates/modals/confirmClaimModifyContact.html'
+        controller: 'claimConfirmModifyEmailCtrl',
+        templateUrl: 'templates/modals/confirmClaimModifyEmail.html',
+        resolve: {
+          email: function () {
+            return $scope.userEmail;
+          }
+        }
+      });
+
+      modalInstance.result.then(function (email) {
+        console.log("new email", email);
+        $scope.userEmail = email;
+      }, function () {
+        console.log('modal-component dismissed at: ' + new Date());
       });
     }
 
-    $scope.onClickConfirm = function() {
-      var claim = claimService.getIncompleteClaim();
-      if (!_.isEmpty(claim)) {
-        net.submitClaim($stateParams.claimId).then(function(resp) {
+    $scope.onClickConfirm = function () {
+      net.submitClaim($stateParams.claimId,
+        {
+          toAddress: $scope.vaAddress,
```

Comment:
technically we are sending to two places: the VA, and a copy to the user's address. Then we are emailing a copy to the user's email.

Now that I think about it, as a service, when we send a letter to the VA, I'm not sure the return address should be a user's address. It should be us. If we screwed up somehow, the letter should go back to us so we can fix it. Having it go back to the user doesn't help since he/she is less likely able to do stuff about it AND already has a copy anyway. If we need the user to fix something, we should email them through our app with instructions instead of hoping that they will fix things themselves.

The return address for both of the letters should be Vetafi since we are acting as a middle ground between the user and the VA. When we send a letter to the VA, the title could even be something like: "From: Vetafi: Aaron Hsu" if we really wanted to show on the letter who it is for. Same thing for a user. If we send a copy to the user, they should know it's from us.

So from this perspective, this body could just be:
```
net.submitClaim(claimId,
  {
    addresses: [vaAddress, userAddress],
    emails: [user.email]  
  }
)
```
This makes the endpoint generic (as mentioned in an earlier comment) since we just have a list of addresses and emails and whatever letters/emails we send, all parties know it's from us.